pkg/segment/results/blockresults/blockresult.go

```diff
@@ -934,33 +924,12 @@ func (rb *RunningBucketResultsJSON) Convert() (*RunningBucketResults, error) {
 		currStats: rb.CurrStats,
 	}
 	currRunningStats := make([]runningStats, 0, len(rb.RunningStats))
-	for statsIdx, rs := range rb.RunningStats {
-		if rb.CurrStats[statsIdx].MeasureFunc == utils.Cardinality {
-			hllString, ok := rs.(string)
-			if !ok {
-				log.Errorf("RunningBucketResultsJSON.Convert: failed to convert hll to string, hll: %+v of type %T", rs, rs)
-				return nil, fmt.Errorf("RunningBucketResultsJSON.Convert: failed to convert hll to byte array")
-			}
-			hllBytes, err := base64.StdEncoding.DecodeString(hllString)
-			if err != nil {
-				log.Errorf("RunningBucketResultsJSON.Convert: failed to decode hllString, err: %v", err)
-				return nil, err
-			}
-			hll, err := structs.CreateHllFromBytes(hllBytes)
-			if err != nil {
-				log.Errorf("RunningBucketResultsJSON.Convert: failed to unmarshal hllBytes, err: %v", err)
-				return nil, err
-			}
-			currRunningStats = append(currRunningStats, runningStats{hll: &toputils.GobbableHll{Hll: *hll}})
-		} else {
-			newVal := utils.CValueEnclosure{}
-			err := newVal.ConvertValue(rs)
-			if err != nil {
-				log.Errorf("RunningBucketResultsJSON.Convert: failed to convert value, err: %v", err)
-				return nil, err
-			}
-			currRunningStats = append(currRunningStats, runningStats{rawVal: newVal})
+	for _, rs := range rb.RunningStats {
+		runningStat, err := rs.GetRunningStats()
+		if err != nil {
+			return nil, err
 		}
```

Comment:
nit: we probably want to log this error

Reply:
We are logging that error [here](https://github.com/siglens/siglens/blob/cebd74e112f079d42ecc63eee07a44b6b04541cb/pkg/segment/results/segresults/segresults.go#L432). It could cause duplicate logs so, i didn't do it. Whats your opinion?

Reply:
addressed in the latest commit.